include/UniTensor.hpp

```diff
@@ -1171,28 +1171,30 @@ namespace cytnx {
       cytnx_error_msg(this->_blocks.size() == 0, "[ERROR][internal] empty blocks for blockform.%s",
                       "\n");
   #endif
-      return this->_blocks[0].dtype();
+      return this->_blocks.size() < 1 ? N_Type + 1 : this->_blocks[0].dtype();
     };
     int device() const {
   #ifdef UNI_DEBUG
       cytnx_error_msg(this->_blocks.size() == 0, "[ERROR][internal] empty blocks for blockform.%s",
                       "\n");
   #endif
-      return this->_blocks[0].device();
+      return this->_blocks.size() < 1 ? 404 : this->_blocks[0].device();
```

Comment:
Do we assume the device value larger and equal to 0 is a GPU device?

Reply:
Yes. See Device.hpp